### DISCUSSION TITLE
Trouble getting started-- Is there no docker image?

### DISCUSSION THREAD

**Discussion by insanity54** [andydunstall/piko#43] [General]
I'm looking to use piko to help with local app development. I develop locally on kubernetes, using a kind cluster with Tilt (for live-reloads, etc.) My app backend is Strapi which handles users accounts and oauth2. The problem I have is that I can't end-to-end test Oauth2 logins unless I have a world-accessible URL for the callback.

piko seems like it could help me by giving world-accessible URLs to services in my development cluster. This way, I could create some domains like strapi.piko.example.com, and frontend.piko.example.com and those domains would route to my piko server, which would reverse proxy requests to the upstream containers running in my local dev cluster.

In my testing, I found https://github.com/andydunstall/piko/tree/main/docs/demo which has a helpful docker-compose.yaml. However when pulling the images, `piko` is missing.

```
docker compose up                                                                                                                                               ✔ 
WARN[0000] /home/cj/Documents/piko/docs/demo/docker-compose.yaml: `version` is obsolete 
[+] Running 5/6
 ⠸ grafana Pulling                                                                                                                                                                                            2.4s 
 ✘ prometheus Error    context canceled                                                                                                                                                                       2.4s 
 ✘ piko-1 Error        context canceled                                                                                                                                                                       2.4s 
 ✘ piko-3 Error        pull access denied for piko, repository does not exist or may require 'docker login': denied: requested access to the resource is denied                                               2.4s 
 ✘ load-balancer Error context canceled                                                                                                                                                                       2.4s 
 ✘ piko-2 Error        context canceled                                                                                                                                                                       2.4s 
Error response from daemon: pull access denied for piko, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

I looked on https://hub.docker.com/search?q=piko and I don't see anything that matches.

I looked on https://artifacthub.io/packages/search?ts_query_web=piko&sort=relevance&page=1 as well, and I don't see anything either.

Is there no docker image for a reason?

**Comment by andydunstall** [comment]
Hi @insanity54, thanks for sharing what your using Piko for

No there is not yet an image on Docker hub, I'll create one this week. For now its best to build the image yourself with `make image` (see [Getting Started](https://github.com/andydunstall/piko/blob/main/docs/getting-started.md))

**Reply by andydunstall** [reply]
Hi @insanity54 I've added Docker images to the GitHub registry for the repo and updated the getting started guide (See https://github.com/andydunstall/piko/pull/59):
- `ghcr.io/andydunstall/piko:v0.2.0`
- `ghcr.io/andydunstall/piko:latest`

Let me know if theres anything else you need :)
